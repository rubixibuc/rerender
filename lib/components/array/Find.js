import React from "react";
import PropTypes from "prop-types";

const render = ({ children, items, predicate }) => {
  if(items && items.length > 0) {
    const found = items.find(predicate);

    if (found) {
      return children(found);
    }
  }

  return null;
}

render.propTypes = {
  children: PropTypes.func.isRequired,
  predicate: PropTypes.func.isRequired,
  items: PropTypes.array
}

render.displayName = "Rerender.Find";

export default render;

