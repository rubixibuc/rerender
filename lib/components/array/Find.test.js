import React from "react";
import Find from "./Find";
import renderer from "react-test-renderer";

test("should render nothing when items is undefined", () => {
  const component = renderer.create(
    <Find predicate={() => true} items={undefined}>
      {item => <div>{item}</div>}
    </Find>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render nothing when items is null", () => {
  const component = renderer.create(
    <Find predicate={() => true} items={null}>
      {item => <div>{item}</div>}
    </Find>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render empty div when items is empty", () => {
  const component = renderer.create(
    <Find predicate={() => true} items={[]}>
      {item => <div>{item}</div>}
    </Find>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render empty div if no items match", () => {
  const component = renderer.create(
    <Find predicate={num => num % 2 === 0} items={[1, 3]}>
      {item => <div>{item}</div>}
    </Find>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render first item that matches", () => {
  const component = renderer.create(
    <Find predicate={num => num % 2 === 0} items={[1, 2, 3, 4]}>
      {item => <div>{item}</div>}
    </Find>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
