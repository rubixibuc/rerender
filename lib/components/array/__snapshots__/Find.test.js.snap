// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render first item that matches 1`] = `
<div>
  2
</div>
`;

exports[`should render nothing if no items match 1`] = `null`;

exports[`should render nothing when items is empty 1`] = `null`;

exports[`should render nothing when items is null 1`] = `null`;

exports[`should render nothing when items is undefined 1`] = `null`;
